bids/analysis/base.py

```diff
@@ -266,11 +266,11 @@ def get_Xy(self, drop_entities=True, **selectors):
         for name, g in data.groupby(ent_cols):
             ent_data = g[ent_cols].drop_duplicates().iloc[0, :]
             ents = ent_data.to_dict()
-            if 'run' in ent_cols:
-                img = self.analysis.layout.get(return_type='file', type='bold',
+            if 'onset' in data.columns:
```

Comment:
I believe this condition is "is first-level", and I think this check does the same job without assuming that we have multiple runs for a given task.